work/builder/builder_test.go

```diff
@@ -327,6 +420,27 @@ func TestPopTxs(t *testing.T) {
 			bundles:         []*Bundle{bundles[0], bundles[1]},
 			expectedTxs:     NewTxOrGenList(txs[1], txs[2], txs[3], txs[4], txs[5], txs[6]),
 		},
+		{
+			name:            "Two bundles - target is popped",
+			incorporatedTxs: NewTxOrGenList(g1, txs[1], txs[2], g3, txs[3], txs[4], txs[5], txs[6]),
+			numToPop:        3,
+			bundles:         []*Bundle{bundles[3], bundles[6]},
+			expectedTxs:     NewTxOrGenList(txs[4], txs[5], txs[6]),
```

Comment:
Shouldn't the expected result be `[txs[5], txs[6]]`?
```
1st pop: g1, txs[1], txs[2], g3 // because g3 requires txs[2]
2nd pop: txs[3]
3rd pop: txs[4]
```